### ISSUE TITLE
bug(autograd): conv2d backward kernel grads vanish in deep stacks; train_autograd plateaus at chance loss

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5465]
## Summary

The tape-based autograd substrate landed in #5457 produces correct gradients on tiny isolated conv2d tests, but in deeper stacks (conv → relu → flatten → linear → CE) the conv kernel gradients are effectively zero (~1e-8 vs ~1e-1 for FC layers in the same network). As a result, the LeNet autograd port from #5462 trains to chance accuracy (~2.76% on EMNIST = 1/47 classes, loss flatlines at ln(47)=3.852) over 5 epochs.

## Evidence

- **Empirical**: `pixi run mojo run -I . examples/lenet_emnist/run_train_autograd.mojo --epochs 5 --batch-size 32 --lr 0.001` produces test accuracy 2.76% vs 74.4% for the manual-grad baseline at epoch 5. Loss starts at ~4.0, decreases briefly to ~3.85, then flatlines.
- **Isolation**: A minimal diagnostic confirmed `loss.backward()` populates gradient registry entries for all conv kernels (no missing entries), but the gradient *magnitudes* are ~7 orders of magnitude smaller than the FC layer gradients in the same network.
- **Test that catches it**: New file `tests/projectodyssey/autograd/test_autograd_convergence.mojo` (PR forthcoming) — `test_conv_linear_softmax_converges` reaches loss=0.69 ≈ ln(2) after 30 steps and plateaus, exactly the chance-loss signature.
- **Test that misses it**: Existing `test_variable_layers.mojo::test_conv2d` passes — it checks one conv layer in isolation with analytical FD comparison on a 3×3 input. The bug only surfaces in deeper stacks.

## Hypothesis (unverified)

Either:
1. `conv2d_backward` in `src/projectodyssey/core/conv.mojo` (or its autograd wrapper in `src/projectodyssey/autograd/backward_ops.mojo`) normalizes the kernel gradient by an extra factor (e.g., batch × spatial × out_channels), reducing it by orders of magnitude.
2. Gradient flow through the second conv layer's input is broken (so c1 sees near-zero upstream grad), even though c2 itself gets a correct grad from the FC layer.

## Reproduction

```bash
# Run the new failing test (red on main):
pixi run mojo run -I src tests/projectodyssey/autograd/test_autograd_convergence.mojo

# Expected fail:
# [test] conv+linear+softmax convergence (30 steps) ...
# Unhandled exception: loss did not decrease enough. first=1.099 last=0.694 drop=0.37 required=0.7
```

## Impact

- `examples/lenet_emnist/train_autograd.mojo` is non-functional on main (merged in #5462).
- Likely affects every example that uses the autograd path with conv layers (issue #5454 sub-tasks for AlexNet, VGG16, ResNet18, MobileNet, GoogLeNet would all hit this).
- Trust in the substrate is compromised — the existing test suite passes while real models fail to train.

## Plan

1. **Tests first** (this PR — forthcoming): land convergence tests that go RED on main, so the bug has a permanent reproducer in CI.
2. **Fix forward** (separate PR): diagnose which of the two hypotheses is correct, fix the `conv2d` backward path, watch the convergence tests go GREEN.
3. **Re-validate LeNet**: run `train_autograd.mojo` for 5 epochs, expect parity with manual-grad baseline (loss ~0.85, test acc ~74%) at epoch 5.

## Related

- Substrate: #5457 (PR), #5452 (issue)
- LeNet port: #5462 (PR), #5461 (issue)
- Tracker for other example ports: #5454 — blocked until this is resolved.
